dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/net/DataOkHttpUploader.kt

```diff
@@ -35,7 +36,12 @@ internal class DataOkHttpUploader(
         batch: List<ByteArray>,
         batchMeta: ByteArray?
     ): UploadStatus {
-        val request = requestFactory.create(context, batch, batchMeta)
+        val request = try {
+            requestFactory.create(context, batch, batchMeta)
```

Comment:
Can we be more explicit in which exception can happen here? It is a code smell to catch any Exception, so we need to have a better understanding of what happens in the request factory.
Also shouldn't this be handled in the request factory directly instead?

Reply:
the idea is that the exception can be `Any` as this can be thrown by an implementation of the `RequestFactory`. You can't know what this will be. We are actually following `iOS` here as they have the same approach on their end. 
You can't handle this in the `RequestFactory` as you need to tell the core what to do with the batch in case of an exception when creating the `Request`